scripts/engine/effect.py

```diff
@@ -158,11 +167,37 @@ def evaluate(self) -> List[Optional[Effect]]:
                 world.recompute_fov(entity)
 
         if success:
-            return self.success_effects
+            return self.success_effects + self.success_triggers
         else:
             return self.failure_effects
 
 
+class TriggerAfflictionsEffect(Effect):
+    def __init__(self, origin: EntityID, target: EntityID, trigger_type: AfflictionTriggerType, success_effects: List[Optional[Effect]],
+            failure_effects: List[Optional[Effect]]):
+
+        super().__init__(origin, success_effects, failure_effects)
+
+        self.trigger_type = trigger_type
+        self.target = target
+
+    def evaluate(self) -> List[Optional[Effect]]:
+        """
+        Trigger all the afflictions on the self.target entity that match the trigger type. Fail if nothing matches
+        """
+        logging.debug(f"Evaluating Trigger Affliction Effect of type '{self.trigger_type}'...")
+        afflictions = world.get_entitys_component(self.target, Afflictions)
+        # iterate over each affliction and trigger it if necessary
+        success = True
+        for affliction in afflictions.active:
+            if self.trigger_type in affliction.triggers:
+                success = success and world.apply_affliction(affliction)
+
+        if success:
+            return self.success_effects
+        return self.failure_effects
+
+
 class AffectStatEffect(Effect):
```

Comment:
I don't think I am following you. AffectStat wasn't modified in this PR

Reply:
That's why I asked! It's the only part of Afflictions with my fingerprints on it so I wondered if you saw any issues with the approach.

Reply:
Gotcha, I think its fine. 

However if I would have to redo it, I would had used a new class (StatModifier) for the stat_modifiers dict instead of a `Tuple[PrimaryStatType, int]`. Why? 

Because it's more extensible. For example suppose that later on I want to make stat modifiers to affect PrimaryStatTypes and SecondaryStatTypes, in the first case I would have to change the dictionary to be of type `Dict[str, Tuple[PrimaryStatType, int, SecondaryStatType, int]]` but that would also break everyplace that's accessing that code and expecting a `Tuple[PrimaryStatType, int]`. In the second case I would just have to add some new fields to the class StatModifier and nothing would break.

Another change I would make is move this code:
```
        if self.cause_name not in afflictions.stat_modifiers:
            afflictions.stat_modifiers[self.cause_name] = (self.stat_to_target, self.affect_amount)
```
into a function (named 'add_stat_modifier' or something) that's inside the Afflictions class. Why? Because if you later on want to change the internal structure of the Afflictions class for some reason you would have to change it on every place thats using `stat_modifiers`

Reply:
That's really helpful, thank you!! I'll make those changes.